flake.nix

```diff
@@ -0,0 +1,24 @@
+{
+  description = "rhine";
+  nixConfig.bash-prompt = "\[rhine\]$ ";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    haskell-flake-utils.url = "github:ivanovs-4/haskell-flake-utils";
+    haskell-flake-utils.inputs.nixpkgs.follows = "nixpkgs";
+    haskell-flake-utils.inputs.flake-utils.follows = "flake-utils";
+
+  };
+
+outputs = { self, nixpkgs, flake-utils, haskell-flake-utils, ... }@inputs:
+  flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-darwin" ] (system:
```

Comment:
Why exactly these systems?

Reply:
Another approach is to [use `eachDefaultSystem`](https://github.com/numtide/flake-utils/blob/master/default.nix#L3-L9) which is more comprehensive, but I went the route of picking the 2 architectures that I was able to test with.

Reply:
I'd propose either to use `eachDefaultSystem` or all systems we can somehow check on CI.

Reply:
> I'd propose either to use eachDefaultSystem or all systems we can somehow check on CI.

If that's OK, I would ask that we relax that requirement given that `aarch64-darwin` (M1 chip) is not available on Github Runners, but that's the machine architecture I'm using. 

What I can do is setup a CI matrix for systems that are available in Github that will build `x86_64-linux` and `x86_64-darwin`  (maybe try `i686-linux` too).

Note that with [this PR](https://github.com/ivanovs-4/haskell-flake-utils/pull/4) `haskell-flake-utils` changed their API such that now we pass an array of systems instead of using `flake-utils.lib.eachSystem` (which is now called in `haskell-flake-utils`).